app/controllers/merchants_controller.rb

```diff
@@ -0,0 +1,56 @@
+class MerchantsController < ApplicationController
+  before_action :require_user, only: [:edit, :show]
+  before_action :redirect_if_logged_in, only: [:new, :create]
```

Comment:
Nice job of using a before action! It can really help DRY up code and when you make changes you have to do it in less places because there isn't a bunch of copy and paste/ find and replace stuff to do.